stringfuzz/transformers/stem_cut.py

```diff
@@ -0,0 +1,128 @@
+'''
+The stem-cut transform picks a subtree and a leaf at random
+and swaps them for each type. Almost like cutting a stem and planting it 
+somewhere else. Names are hard.
+'''
+
+import random
+
+from stringfuzz.ast import *
+from stringfuzz.ast_walker import ASTWalker
+from stringfuzz.generators import random_text
+from stringfuzz.parser import parse
+
+__all__ = [
+    'stem_cut',
+]
+
+class StemCutTransformer(ASTWalker):
+    def __init__(self, ast, pairs):
+        super(StemCutTransformer, self).__init__(ast)
+        self.pairs = pairs
+
+    @property
+    def ast(self):
+        return self._ASTWalker__ast
+
+    def enter_expression(self, expr):
+        for i in range(len(expr.body)):
+            for pair in self.pairs:
+                if expr.body[i] == pair[0]:
+                    expr.body[i] = pair[1]
+                elif expr.body[i] == pair[1]:
+                    expr.body[i] = pair[0]
+
+class StemCutFinder(ASTWalker):
+    def __init__(self, ast):
+        super(StemCutFinder, self).__init__(ast)
+        #            expr, lit
+        self.str  = [None, None]
+        self.bool = [None, None]
+        self.int  = [None, None]
+        self.rx   = [None, None]
```

Comment:
Does this transformer perform at most 1 graft for each type?

Reply:
Yup!